pkg/spdx/implementation.go

```diff
@@ -673,13 +682,32 @@ func (di *spdxDefaultImplementation) PackageFromImageTarball(
 
 	logrus.Infof("Image manifest lists %d layers", len(manifest.LayerFiles))
 
-	// Cycle all the layers from the manifest and add them as packages
+	// Scan the container layers for OS information:
+	ct := osinfo.ContainerScanner{}
+	layerPaths := []string{}
 	for _, layerFile := range manifest.LayerFiles {
+		layerPaths = append(layerPaths, filepath.Join(tarOpts.ExtractDir, layerFile))
+	}
+	layerNum, osPackageData, err := ct.ReadOSPackages(layerPaths)
+	if err != nil {
+		return nil, errors.Wrap(err, "getting os data from container")
+	}
+
+	if osPackageData != nil {
```

Comment:
Returning a non-pointer slice would help here to avoid the nil pointer checks. Same for the iteration below (L725).

Reply:
Kept the pointer so symbolize non-supported OS (see other comment)